### DISCUSSION TITLE
Additional Information/Parameter on Purchase/Sales Orders

### DISCUSSION THREAD

**Discussion by ghost** [inventree/InvenTree#2867] [Q&A] [answered]
Hi,

first of all thanks for the great SW. We are evaluating InvenTree to use it for a small project. So far personally I like the concept.

For the reporting and the cross-linking to other software, we need to store additional information to each purchase & sales order. For parts there is the possibility to add parameters. Is there a similar functionality for orders?

It does not have the complexity to change the fields for each SO/PO. More like a additional reference field e.g. for shipping contact or internal project number.

Thanks in advance
Jonas

**Comment by matmair** [comment, marked as answer]
Hi there @jonasexpleo !
In general we do not add functions for single-users as that will blow up the application fast. Could you specify how other users would benefit from or use  this addition?
There is always the option to just use a plugin with the AppMixin to add custom models to attach additional data to built-in models or logic.

**Reply by ghost** [reply]
@matmair & @SchrodingersGat Thank you for the responses. I should have explained the use-case a little bit more.

I'm trying to use the sales order report system for printing a delivery note.
In the delivery note we need some additional information:

* for a bigger company we need to define the recipient of the package,
* in some cases a different shipping address (most companies have different locations)

For filtering and ordering of the SO:
* order priority
* a tag like product cluster for filtering the orders

I have considered:

* Using the description field for the information. It would be usable for me, for example with a comma as separation between the data, but there are users with no technical background and it would be easy to miss/mix up the information.
* Using the notes field. But we have in our big overview table already a notes/comment field and I rather want to let the users to append comments.
* Adding for each recipient/address of a company an inventree company. One side this is doable, but it would reduce the possibility to get an overview how many/which orders went to a company. Additionally many recipient-companies would have only one order.

I'm fine using/developing a plugin and in the end it is only a idea, but I positive that it isn't a unusual use-case to store additional information to each order. The main benefits I see:

* The user has the possibility to adapt the field to his needs and no code needs to be changed.
* A parameter like storage has the possibility to reuse the data in the reporting system.
* The structured data has also the possibility to easily change the data through the API.
* This would have also the benefit to easily spot missing data, e.g. with a script.

I'm still new to inventree therefore, I'm still exploring it.

---

Off-Topic: In this use-case it would be good to choose also a shipping for the report or having a report button next to a shipment. We have mostly only one shipment, therefore it should be fine like it is.

**Reply by matmair** [reply]
@jonasexpleo this definitely sounds like a feature request. Please file one so we can assign milestones etc. and track it. We will have to check when we can fit it in as we are a slim team of core devs and mostly focus on what our own respective companies need to be implemented.

**Reply by ghost** [reply]
Created a new issue #3003

**Comment by SchrodingersGat** [comment]
@jonasexpleo could you make use of the existing "description" or "notes" field to achieve this?

**Reply by ghost** [reply]
@SchrodingersGat  This is how I do it in our test system. With more than one/two information it doesn't feel good. Special if you want to reuse the data. Look for the use-case in the other [replay](https://github.com/inventree/InvenTree/discussions/2867#discussioncomment-2640209).
